src/models/discriminant_analysis.jl

```diff
@@ -3,130 +3,123 @@
-    outdim = ifelse(model.outdim == 0, def_outdim, model.outdim)
-    # Check if the given one is sensible
-    if outdim > def_outdim
-        throw(
-            ArgumentError(
-                "`outdim` must not be larger than `min(p, nc-1)`"*
-                "where `p` is the number of features in `X` and"*
-                "`nc` is the number of unique classes in the target vector."
-            )
-        )
-    end
-    return Xm_t, yplain, classes_seen, p, n, nc, nclasses, integers_seen, outdim
+  pool = MMI.classes(y[1]) # Class list containing entries in pool of y.
+  classes_seen = unique(y) # Class list of actual entries in seen in y.
+  nc = length(classes_seen) # Number of classes in pool of y.
+
+  Xm_t = _matrix_transpose(model, X) # Now p x n matrix
+  p, n = size(Xm_t)
+
+  # Check to make sure we have more than one class in training sample.
+  # This is to prevent Sb from being a zero matrix
+  # see issue #41
+  lone_class_unsupported = model isa Union{LDA, BayesianLDA} &&
+        model.cov_b != MS.SimpleCovariance()
+  if nc <= 1 && lone_class_unsupported
+      throw(ERR_LONE_TARGET_CLASS)
+  end
+
+  # Check to make sure we have more samples than classes.
+  # This is to prevent Sw from being the zero matrix.
+  if n <= nc
+      throw(
+          ArgumentError(
+              "The number of training samples `n` has"*
+              " to be greater than number of unique classes `nc`."
+          )
+      )
+  end
+  # Check output dimension default is min(p, nc-1)
+  def_outdim = min(p, nc - 1)
+  # If unset (0) use the default; otherwise try to use the provided one
+  outdim = ifelse(model.outdim == 0, def_outdim, model.outdim)
+  # Check if the given one is sensible
+  if outdim > def_outdim
+      throw(
+          ArgumentError(
+              "`outdim` must not be larger than `min(p, nc-1)`"*
+              "where `p` is the number of features in `X` and"*
+              "`nc` is the number of unique classes in the target vector."
+          )
+      )
+  end
+  return Xm_t, y, classes_seen, pool, p, n, nc, outdim
+end
+
+function MMI.fit(model::LDA, ::Int, X, y)
+  Xm_t, _ , classes_seen, pool, _, _, nc, outdim =
+      _check_lda_data(model, X, y)
+  core_res = MS.fit(
+      MS.MulticlassLDA, Xm_t, y;
+      method=model.method,
+      outdim,
+      regcoef=model.regcoef,
+      covestimator_within=model.cov_w,
+      covestimator_between=model.cov_b
+  )
+  cache = nothing
+  report = (
+      indim = MS.size(core_res)[1],
+      outdim=MS.size(core_res)[2],
+      mean=MS.mean(core_res),
+      nclasses=nc,
+      class_means=MS.classmeans(core_res),
+      class_weights=MS.classweights(core_res),
+      Sb=MS.betweenclass_scatter(core_res),
+      Sw=MS.withclass_scatter(core_res)
+  )
+  fitresult = (core_res, classes_seen, pool)
+  return fitresult, cache, report
 end
 
-function MMI.fitted_params(::LDA, (core_res, classes_seen))
-    return (
-        projected_class_means=MS.classmeans(core_res),
```

Comment:
So user can no longer access projected class means. What's the reason for no longer exposing this training outcome to the user?

Reply:
It's been moved to the report. The LDA models fit parameter should be the projection matrix, all other stuff should be put in the report.